### PR TITLE
Import annotation jobs from `cpg_workflows`

### PR DESCRIPTION
**Pull request by vladsavelyev**
Depends on https://github.com/populationgenomics/production-pipelines/pull/155

# Fixes

  - Fixes https://github.com/populationgenomics/automated-interpretation-pipeline/issues/135

## Proposed Changes

  - Add dependency on `cpg_workflows` package 
  - Import annotation code from `cpg_workflows` instead of duplicating
  - Use `cpg_workflows.batch.get_batch()` to initialise Batch, and move Hail parameters into the config

## Checklist

- [x] Related Issue created
- [ ] Tests covering new change
- [ ] Linting checks pass

### CODE REVIEW

**Comment by MattWellie** [comment]
@vladsavelyev this looks awesome, thanks so much! I was planning to get started on this today ☺️

File: reanalysis/interpretation_runner.py [hunk 1]
```diff
@@ -415,18 +333,30 @@ def main(
...
+        anno_job = annotate_cohort_jobs(
+            b=get_batch(),
+            vcf_path=to_path(input_path),
+            vep_ht_path=to_path(vep_ht_tmp),
+            out_mt_path=to_path(ANNOTATED_MT),
+            checkpoint_prefix=to_path(output_path('annotation_temp', 'tmp')),
+            depends_on=vep_jobs,
         )
-        prior_job.depends_on(*annotation_jobs)
+        prior_job = anno_job[-1]
```
Comment (MattWellie) [inline reanalysis/interpretation_runner.py:359]:
```suggestion
        prior_job = anno_job[-1]
        prior_job.depends_on(*vep_jobs)
```

the OG annotate_cohort_jobs doesn't implement `depends_on`, so this needs to be set separately

File: reanalysis/reanalysis_global.toml [hunk 2]
```diff
@@ -1,7 +1,6 @@
-[buckets]
-web_suffix = "web"
-tmp_suffix = "tmp"
-analysis_suffix = "analysis"
+[workflow]
+name = 'AIP'
+scatter_count = 50
```
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply reanalysis/reanalysis_global.toml:4]: Oh, sorry, I'll look into that! 
Was hoping we could harmonise this with https://github.com/populationgenomics/cpg-utils/pull/117

File: reanalysis/interpretation_runner.py [hunk 3]
```diff
@@ -415,18 +333,30 @@ def main(
...
-        prior_job = annotated_mt_from_ht_and_vcf(
-            input_vcf=input_path, batch=batch, job_attrs={}, vep_ht=vep_ht_tmp
+        # Apply the HT of annotations to the VCF, save as MT
+        anno_job = annotate_cohort_jobs(
+            b=get_batch(),
+            vcf_path=to_path(input_path),
+            vep_ht_path=to_path(vep_ht_tmp),
+            out_mt_path=to_path(ANNOTATED_MT),
+            checkpoint_prefix=to_path(output_path('annotation_temp', 'tmp')),
+            depends_on=vep_jobs,
```
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply reanalysis/interpretation_runner.py:357]: Hmm?

File: reanalysis/interpretation_runner.py (hunk 1 above)
Comment (vladsavelyev) [reply reanalysis/interpretation_runner.py:359]: What do you mean? `annotate_cohort_jobs` would be updated in `cpg-workflows`. `depends_on` is needed for the dataproc version of annotation

File: reanalysis/interpretation_runner.py [hunk 4]
```diff
@@ -415,18 +333,30 @@ def main(
...
-        # apply annotations
-        prior_job = annotated_mt_from_ht_and_vcf(
-            input_vcf=input_path, batch=batch, job_attrs={}, vep_ht=vep_ht_tmp
+        # Apply the HT of annotations to the VCF, save as MT
+        anno_job = annotate_cohort_jobs(
+            b=get_batch(),
+            vcf_path=to_path(input_path),
+            vep_ht_path=to_path(vep_ht_tmp),
+            out_mt_path=to_path(ANNOTATED_MT),
+            checkpoint_prefix=to_path(output_path('annotation_temp', 'tmp')),
```
[the remark this answers is not in the document]
Comment (MattWellie) [reply reanalysis/interpretation_runner.py]: I'd definitely like to coordinate with those changes, but until that change is finalised & merged I think the current code here still needs the `buckets` section to work properly.

Agreed though, https://github.com/populationgenomics/cpg-utils/pull/117 should make this code-infra relationship much easier

File: reanalysis/interpretation_runner.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply reanalysis/interpretation_runner.py:363]: Yeah, but I'm updating `annotate_cohort_jobs` in a parallel PR

File: reanalysis/interpretation_runner.py (hunk 1 above)
Comment (MattWellie) [reply reanalysis/interpretation_runner.py:365]: Ahhhhh, I didn't realise this introduces dataproc as a dependency again - that will be a problem for working with MSFT

Comment (vladsavelyev) [reply reanalysis/interpretation_runner.py:365]: That's optional, there is `use_dataproc` flag. I just find it easier to debug with dataproc

Comment (vladsavelyev) [reply reanalysis/interpretation_runner.py:366]: Updated to make sure use_dataproc=False

File: reanalysis/reanalysis_global.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply reanalysis/reanalysis_global.toml:9]: Reverted that change for now

**Review by MattWellie — APPROVED** [review]
Heroic effort

**Comment by vladsavelyev** [comment]
Thanks for approving! Note that this needs to be merged too: https://github.com/populationgenomics/cpg-utils/pull/119

**Pull request merged**